apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -62,10 +62,22 @@ export default class CdoFieldButton extends GoogleBlockly.Field {
     this.buttonHandler_(this.setValue.bind(this));
   }
 
-  updateWidth_() {
-    super.updateWidth_();
+  updateSize_() {
```

Comment:
What do you think about add a comment when we're overriding functions vs adding new functions to this class?
Also, this is not part of your changes, but I noticed that there is already a `getValue` function defined in `field.ts` so we could remove it from this class.

Reply:
Looking at the google blockly field class, I think these are all overrides technically? Looks like their getValue doesn't guarantee returning a string (if my typescript comprehension is correct?), might want to use getText instead. And I see at least the `setValue` override is not recommended (maybe others as well 😬 ), [link](https://github.com/google/blockly/blob/fa9cb579ae119749ee14277518374060bc555c3e/core/field.ts#L983-L990)

Reply:
Interesting! It seems like the value stored by the field is ensured to be a certain type by the validation done in `setValue` And yeah, I see the recommendation in the `setValue` comment to override `doClassValidation_` and `doValueUpdate_`. Maybe in a later refactor we override `doValueUpdate_` instead of `setValue`?